lib/gollum-lib/filter/sourcemap.rb

```diff
@@ -0,0 +1,24 @@
+# ~*~ encoding: utf-8 ~*~
+
+# Source Maps
+#
+# Find lines corresponding to h1 headers for the present markup and insert markers indicating the line number in the raw source.
+
+class Gollum::Filter::SourceMap < Gollum::Filter
+  def extract(data)
+  	regex = Gollum::Markup.formats[@markup.format][:h1]
+    return data if regex.nil?
+    i = 0
+    data.lines.map do |line|
```

Comment:
Ruby's `gsub' with a block unfortunately doesn't have a line index variable, so we need to split the data, loop over it to get the index and generate the markers, and then join it again. :-(